config/application.php

```diff
@@ -64,14 +64,12 @@
 $table_prefix = env('DB_PREFIX') ?: 'wp_';
 
 if (env('DATABASE_URL')) {
-    $database_dsn = parse_url(env('DATABASE_URL'));
-    $db_name = substr($database_dsn['path'], 1);
-    $db_host = isset($database_dsn['port']) ? $database_dsn['host'] . ":" . $database_dsn['port'] : $database_dsn['host'];
-
-    Config::define('DB_NAME', $db_name);
-    Config::define('DB_USER', $database_dsn['user']);
-    Config::define('DB_PASSWORD', $database_dsn['pass'] ?? null);
-    Config::define('DB_HOST', $db_host);
+    $dsn = (object) parse_url(env('DATABASE_URL'));
+
+    Config::define('DB_NAME', substr($dsn->path, 1));
+    Config::define('DB_USER', $dsn->user);
+    Config::define('DB_PASSWORD', $dsn->pass ?? null);
```

Comment:
@QWp6t didn't you say `?? null` wasn't needed? Or would this result in `""` which would actually be used as the password instead of "skipped"?

Reply:
I had kept it from the original implementation but it does look like it would be fine to remove it as `$dsn->pass` is returning `NULL` and doesn't appear to be throwing any errors when testing with:

```
DATABASE_URL=mysql://root@localhost/wp_sage
```

while having the DB_ vars commented out.

Reply:
and that's why i'm not a fan of forcing an array into an object via casting. Now you're accessing properties that may not exist in the case you posted in your comment and it will trigger warnings.

```
DATABASE_URL=mysql://root@localhost/wp_sage
```

https://3v4l.org/Yf3ck

Working with objects is great but only if the source was crafted/thought as an object ;)

Reply:
Good catch, thank you! So `?? null` does this fix. Should be noted that the array would behave the same anyway? Either way we need `??`

Reply:
yeah in both cases we need it to handle passwordless db

Reply:
Yeah, I was going to post an update to my above– I didn't push any changes as I assumed there was a "gotcha" somewhere. My PHP wasn't throwing errors locally but I assumed it *should*.

And yes, for the record, an array would behave the same way. https://3v4l.org/LTTEX – it has nothing to do with casting to an object. :)